src/js/base/Pagination.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import { Button } from "../base";
+import styled from "styled-components";
+import { range } from "lodash-es";
+
+const StyledPaginationBox = styled.div`
+    justify-content: center;
+    display: flex;
+    width: 500px;
+    margin-left: 25%;
+    margin-right: 25%;
+    button:first-of-type {
+        width: 100px;
+    }
+    button:last-child {
+        width: 100px;
+    }
+
+    button {
+        margin: 5px;
+        height: 25px;
+    }
+`;
+
+export const sliceData = (items, currentPage, itemsPerPage) => {
+    const indexOfLastItem = currentPage * itemsPerPage;
+    const indexOfFirstItem = indexOfLastItem - itemsPerPage;
+    return items.slice(indexOfFirstItem, indexOfLastItem);
+};
+
+const getPagesToDisplay = (pageNumbers, currentPage) => {
+    if (pageNumbers.length <= 4) {
+        return pageNumbers;
+    } else if (currentPage === 1) {
+        return pageNumbers.slice(0, 4);
+    } else if (currentPage === pageNumbers.length) {
+        return pageNumbers.slice(pageNumbers.length - 4, pageNumbers.length);
+    } else if (currentPage === pageNumbers.length - 1) {
+        return pageNumbers.slice(currentPage - 3, currentPage + 1);
+    }
+    return pageNumbers.slice(currentPage - 2, currentPage + 2);
+};
+
+export const Pagination = ({ itemsPerPage, totalItems, color }) => {
+    const [currentPage, setCurrentPage] = useState(1);
+    const paginate = number => setCurrentPage(number);
+
+    const pageNumbers = range(1, Math.ceil(totalItems / itemsPerPage + 1));
```

Comment:
Are you sure defining this array is the most efficient and clear way to implement this?